### ISSUE TITLE
For CampaignManager, EventManager and TeamLead why are we not using Claims?

### ISSUE THREAD

**Issue by mgmccarthy** [HTBox/allReady#2003]
@stevejgordon @tonysurma @MisterJames 

From my understanding, we're keeping the id's of the entity(ies) whose type a user is a manager of in the database, which is fine.

However, there are checks supported by UserAuthorizationService that return whether or not a given user is a CampaignManager (in which one or more Camapaign Id's are found in the CampaignManagers table)...

instead of a db query (and I believe there is a cache invovled here), these type of directives seem to belong to Claims instead.

as far as the actual id's of the entity the user might me a manager of, those still should be kept in the database.

this is obviously a question, and I'm most likely not understanding the full picture here, but I thought I'd ask.

**Comment by stevejgordon** [comment]
Hi Mike.

It was a combination of things but one of the main ones being the planned move to Identity Server. At the time when that moves forward we won't want the identity server to have specific knowledge of the authorization within the client sites (allReady). So the IdentityServer will authenticate the user only.

At that point we will need some kind of lookup on allReady to get the user local authorization (event manager etc) and apply it as we are now. Claims are an option, but they still live per request and even if we use ASP.NET Core Identity, under the hood that simply stores claims in a DB table and retrieves them.

It also depends on Identity which I think we can drop out when we move to IdentityServer.

I think that's the main reason. I haven't added as much caching in there as I planned to yet as wanted to see how it performed before optimising. It feels like a 5-10 min cache on those lookups would be reasonable though.

**Comment by mgmccarthy** [comment]
@stevejgordon thanks for the answer. Since I know nothing about Identity Server (what is the specific Identity Server we're looking for?), sounds like we'll keep the code as-is for now.

Do you happen to know which "Identity Server" we'll be using? This one? http://docs.identityserver.io/en/release/#

**Comment by stevejgordon** [comment]
Yep, that's the one. We have a repo for it but put it hold while working on the v1 milestone for allReady.
